roadrunner.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+# Namespace for all RoadRunner code
+module RoadRunner
+  def self.initialize_report_data
+    report_data = {}
+    report_data['productId'] = 'AWS SDK for Ruby v3'
+    begin
+      report_data['commitId'] = `git rev-parse HEAD`.strip
+    rescue StandardError
+      # unable to get a commit, maybe run outside a git repo.  Skip
+    end
+    report_data['results'] = []
+    report_data
+  end
+
+  def self.convert_result(service, data, date, dimensions)
+    results = []
+    service_name = service.split('-')[-1]
+    data.each do |key, value|
+      next if key == 'gem_version'
+
+      result = {}
+      result['name'] = "#{service_name}.#{generate_result_name(key)}"
+      result['description'] = generate_description(key)
+      result['unit'] = unit(key)
+      result['date'] = date
+      result['dimensions'] = generate_dimensions(key, dimensions)
+      result['measurements'] = value.is_a?(Array) ? value : [value]
+
+      # RoadRunner currently doesn't support kilobyte units. Converting to megabytes instead.
+      if result['unit'] == 'Kilobytes'
+        result['unit'] = 'Megabytes'
+        if result['measurements'].is_a?(Array)
+          result['measurements'].each_with_index do |measurement, index|
+            result['measurements'][index] = measurement / 1000.0
+          end
+        else
+          result['measurements'] = result['measurements'] / 1000.0
+        end
+      end
+      results << result
+    end
+    results
+  end
+
+  def self.generate_result_name(benchmark)
+    split = benchmark.split('_')
+    split.delete('mem')
```

Comment:
Will these strings guarantee to exist? I think a better approach would be to be less "magical"/generic and more "boring"/explicit here - I think check for benchmark names instead of trying to parse out keywords from them. This implementation is incredibly dependent on the string format and it's not clear how. You could even raise or warn on unhandled ones.